core/templates/expressions/expression-syntax-tree.service.ts

```diff
@@ -16,417 +16,475 @@
+        this.verifyNumArgs(args, 2);
+        return args[0] === args[1];
       },
-      '==': {
-        eval: function(args) {
-          verifyNumArgs(args, 2);
-          return args[0] === args[1];
-        },
-        getType: function(args) {
-          verifyNumArgs(args, 2);
-          return PARAMETER_TYPES.UNICODE_STRING;
-        }
+      getType: (args: Array<string>): string => {
+        this.verifyNumArgs(args, 2);
+        return AppConstants.PARAMETER_TYPES.UNICODE_STRING;
+      }
+    },
+    '!=': {
+      eval: (args: Array<string>): boolean => {
+        this.verifyNumArgs(args, 2);
+        return args[0] !== args[1];
       },
-      '!=': {
-        eval: function(args) {
-          verifyNumArgs(args, 2);
-          return args[0] !== args[1];
-        },
-        getType: function(args) {
-          verifyNumArgs(args, 2);
-          return PARAMETER_TYPES.UNICODE_STRING;
-        }
+      getType: (args: Array<string>): string => {
+        this.verifyNumArgs(args, 2);
+        return AppConstants.PARAMETER_TYPES.UNICODE_STRING;
+      }
+    },
+    '&&': {
+      eval: (args: Array<string>): boolean => {
+        this.verifyNumArgs(args, 2);
+        return Boolean(args[0] && args[1]);
       },
-      '&&': {
-        eval: function(args) {
-          // TODO(kashida): Make this short-circuit.
-          verifyNumArgs(args, 2);
-          return Boolean(args[0] && args[1]);
-        },
-        getType: function(args) {
-          verifyNumArgs(args, 2);
-          _verifyArgTypesMatchExpectedType(
-            args, PARAMETER_TYPES.UNICODE_STRING);
-          return PARAMETER_TYPES.UNICODE_STRING;
-        }
+      getType: (args: Array<string>): string => {
+        this.verifyNumArgs(args, 2);
+        this.verifyArgTypesMatchExpectedType(args,
+          AppConstants.PARAMETER_TYPES.UNICODE_STRING);
+        return AppConstants.PARAMETER_TYPES.UNICODE_STRING;
+      }
+    },
+    '||': {
+      eval: (args: Array<string>): boolean => {
+        this.verifyNumArgs(args, 2);
+        return Boolean(args[0] || args[1]);
       },
-      '||': {
-        eval: function(args) {
-          // TODO(kashida): Make this short-circuit.
-          verifyNumArgs(args, 2);
-          return Boolean(args[0] || args[1]);
-        },
-        getType: function(args) {
-          verifyNumArgs(args, 2);
-          _verifyArgTypesMatchExpectedType(
-            args, PARAMETER_TYPES.UNICODE_STRING);
-          return PARAMETER_TYPES.UNICODE_STRING;
-        }
+      getType: (args: Array<string>): string => {
+        this.verifyNumArgs(args, 2);
+        this.verifyArgTypesMatchExpectedType(args,
+          AppConstants.PARAMETER_TYPES.UNICODE_STRING);
+        return AppConstants.PARAMETER_TYPES.UNICODE_STRING;
+      }
+    },
+    // Note that removing quotation marks from this key causes issues with
+    // minification (when running the deployment scripts).
+    /* eslint-disable quote-props */
```

Comment:
Is this comment still relevant? Looks like you're able to remove the quotation marks, so I guess not?

Reply:
That's because `if` is a reserved key which linter tells as well.

Reply:
👍